### PR TITLE
Transfer create query

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Use a query for creating transfers, so that a newly created query gets added to the landing tables. There's a small bug I'm still trying to work out (see below).

### CODE REVIEW

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/EntityTransfersCreate.tsx [hunk 1]
```diff
@@ -58,18 +52,12 @@ export const EntityTransfersCreate: React.FC<{}> = _ => {
...
   const handleCreateTransfer = (payload: CreateTransferPayload) => {
-    setErrors(undefined);
-    setCreating(true);
-    createEntityTransfer(payload)
-      .then(transfer => {
-        // Transfer is the new transfer object; send it off to the modal.
-        setCreating(false);
+    createTransfer(payload, {
+      onSuccess: transfer => {
+        queryClient.invalidateQueries('entity-transfers');
```
Comment (Jskobos) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/EntityTransfersCreate.tsx:57]: This seems like it should do what we want, but it doesn't. To check:
1. Have a pending transfer (or several) already.
1. Navigate directly to the create page (don't use the button, or else you can reload the page). 
2. Create a transfer
3. Observe: the new transfer is the only one in the "Pending" list

Comment (Jskobos) [reply packages/manager/src/features/EntityTransfers/EntityTransfersCreate/EntityTransfersCreate.tsx:57]: EDIT: fixed this with help from John. All should be well.

File: packages/manager/src/queries/entityTransfers.ts [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
     getAllEntityTransfersRequest,
     queryPresets.longLived
   );
 };
+
+export const useTransferQuery = (token: string, enabled: boolean = true) => {
+  return useQuery<EntityTransfer, APIError[]>(
+    [queryKey, token],
+    () => getEntityTransfer(token),
+    { ...queryPresets.shortLived, enabled, retry: false }
```
Comment (johnwcallahan) [inline packages/manager/src/queries/entityTransfers.ts:32]: re: retry; this was bothering me too. I wonder if a more comprehensive solution is to _not_ retry on 404s. Something to think about for later.

**Review by johnwcallahan — APPROVED** [review]

**Review by tiffwong — APPROVED** [review]
The pending transfers seem to display as expected 👍

**Review by DevDW — APPROVED** [review]
Working well 👌

**Pull request merged**